### PR TITLE
Update interview details for distribution

### PR DESCRIPTION
**Pull request by pecigonzalo**
- Add Kubernetes exercise description.
- Use local Kubernetes clusters.

I believe using a local cluster will be an easier way for candidates to provide code that we can replicate locally. The exercise will mention that this should be treated as a remote environment.

### CODE REVIEW

File: handbook/engineering/hiring/software-engineer-distribution.md [hunk 1]
```diff
@@ -30,15 +30,15 @@ Learn more about what it is like to work at Sourcegraph by reading [our handbook
 1. We set up a 30-minute call to learn more about what you are looking for, tell you about Sourcegraph, and answer any questions that you have.
 1. We evaluate relevant technical skills that you have via an asynchronous coding exercise.
    - We will give you an overview of the exercise in advance.
-   - We will send you the details at a time of your choosing and you will have up to 2 hours to work on the exercise.
+   - We will send you the details at a time of your choosing and you will have up to 2 hours to work on each of the two exercises.
    - You will be able to use your own development environment and lookup documentation on the internet.
-1. We schedule 4 hours of remote interviews over video chat across multiple days.
+2. We schedule 4 hours of remote interviews over video chat across multiple days.
```
Comment (pecigonzalo) [inline handbook/engineering/hiring/software-engineer-distribution.md:35]:
```suggestion
1. We schedule 4 hours of remote interviews over video chat across multiple days.
```

File: handbook/engineering/hiring/software-engineer-distribution.md [hunk 2]
```diff
@@ -30,15 +30,15 @@ Learn more about what it is like to work at Sourcegraph by reading [our handbook
...
    - **Architecture:** We give you an open problem statement and you walk us through how you would solve the problem.
    - **Technical experience:** We ask you about your past work and accomplishments.
    - **Team collaboration:** We ask you about how you work and communicate in a team setting, and how you handle tricky situations.
    - **CTO:** We ask you about what motivates you to do your best work, and we tell you more about the vision for the company.
-1. We check your references.
-1. We make you a job offer.
+3. We check your references.
```
Comment (pecigonzalo) [inline handbook/engineering/hiring/software-engineer-distribution.md:40]:
```suggestion
1. We check your references.
```

File: handbook/engineering/hiring/software-engineer-distribution.md [hunk 3]
```diff
@@ -30,15 +30,15 @@ Learn more about what it is like to work at Sourcegraph by reading [our handbook
...
+4. We make you a job offer.
```
Comment (pecigonzalo) [inline handbook/engineering/hiring/software-engineer-distribution.md:41]:
```suggestion
1. We make you a job offer.
```

**Comment by davejrt** [comment]
For the sake of documenting correctly, I proposed to @pecigonzalo that two tests, 4 hrs total seems like a bit much to ask of some people. We agreed that perhaps combining to two tests, to be 3 hrs, and have some coding component along with the existing infra/k8s work was more reasonable for candidates.

File: handbook/engineering/hiring/software-engineer-coding-exercise.md [hunk 4]
```diff
@@ -65,6 +65,19 @@ We designed [this synchronous exercise](https://github.com/sourcegraph/interview
...
 If we device to move forward, we will schedule interviews with team members.
 
+## Kubernetes exercise
+
+We designed this exercise to measure your skills at integrating build and deployment workflows into a development lifecycle.
+
+- You will choose a two hour timeframe to independently work on the exercise. You can choose any day and time that works for you; there are no scheduling constraints.
+- At your chosen date and time, we will send you detailed instructions. You will be building a simple pipeline to build and deploy a set of Go services.
+- You should have a recent version of Go (≥ 1.13) installed on your computer.
+- You should have a simple Kubernetes local cluster ([kind](https://kind.sigs.k8s.io/), [k3d](https://github.com/rancher/k3d), [minikube](https://minikube.sigs.k8s.io/docs/start/))
```
Comment (nicksnyder) [inline handbook/engineering/hiring/software-engineer-coding-exercise.md:75]: Did you verify that the yaml works on all of these? PVs behave differently, which may impact the solution.

**Review by nicksnyder — APPROVED** [review]

File: handbook/engineering/hiring/software-engineer-coding-exercise.md (hunk 4 above)
Comment (pecigonzalo) [reply handbook/engineering/hiring/software-engineer-coding-exercise.md:75]: I tested `kind` and `k3d`, but minikube should also work as I believe it has a stateful volume. Ill merge this as I have an email pending on it and make sure to verify and update before I send the actual test.

**Comment by pecigonzalo** [comment]
@davejrt I have updated this to describe a 3h test that combines both things and clarify the languages section as well. How does this look to you?

Im adapting part of t he codehost test to be a bit simpler.

**Review by davejrt — APPROVED** [review]
> @davejrt I have updated this to describe a 3h test that combines both things and clarify the languages section as well. How does this look to you?
> 
> Im adapting part of t he codehost test to be a bit simpler.

LGTM

File: handbook/engineering/hiring/software-engineer-coding-exercise.md (hunk 4 above)
Comment (pecigonzalo) [reply handbook/engineering/hiring/software-engineer-coding-exercise.md:75]: - [x] `kind`
- [x] `k3d`
- [ ] `minikube`

**Pull request merged**